gufe/tokenization.py

```diff
@@ -77,6 +78,17 @@ def _gufe_tokenize(self):
         """
         return normalize(self.to_keyed_dict(include_defaults=False))
 
+    @property
+    def logger(self):
+        """Return logger named by this module and class qualname"""
+        if (logger := getattr(self, '_logger', None)) is None:
+            cls = self.__class__
+            logname = cls.__module__ + "." + cls.__qualname__
```

Comment:
Per-instance logging instead of per-class? It's not impossible, but it would mean a lot more loggers. I'm not sure how much overhead that would add.

Note that ["Multiple calls to getLogger() with the same name will always return a reference to the same Logger object."](https://docs.python.org/3/library/logging.html#logger-objects) So in the current implementation we do not create one logger instance per instance, but one logger instance per class and all members of the class have  pointer to it.

(`@classmethod` and `@property` don't play well together, otherwise I would have implemented this as a class-property. I decided keeping it a property is more important than making it clear that this is a class property -- it's already documented in the docs for getLogger.)

Reply:
Its more looking at the test cases, once we've got more than one Unit running we'll want to be able to easily go from logs to work unit. Maybe this can be solved through storing one set of logs per Unit though, so the file name is the Key.

Reply:
FWIW, having this here in the superclass makes it easy to change that in the future if we want. Maybe we go with this middle road now, and can check the performance impact of adding more fine details later? (We can also drop down to module-only if we find that per-class is an issue.)

Reply:
> Its more looking at the test cases, once we've got more than one Unit running we'll want to be able to easily go from logs to work unit. Maybe this can be solved through storing one set of logs per Unit though, so the file name is the Key.

Or by defining `ProtocolUnit.logger` and overriding this.